src/System.Security.Cryptography.Csp/tests/CspParametersTests.cs

```diff
@@ -33,5 +34,103 @@ public static void SetFlags_ValidatesInput()
+            }
+        }
+
+        [Theory(Skip = "Manual test - requires Smart Card - read instructions")]
+        [InlineData(true)]
+        [InlineData(false)]
+        public static void KeyPassword_SmartCard_Manual_Test(bool correctPassword)
+        {
+            // Find info about your smart card:
+            // > certutil -scinfo -silent
+            //   You should see something like:
+            //   Provider = Microsoft Base Smart Card Crypto Provider
+            //   Key Container = {123ABCDE-1234-ABCD-1234-ABCD1234ABCD}
+            // Fill consts below
+            // Run this test - enter correct/incorrect password for your smart card when asked
+            const string provider = "Microsoft Base Smart Card Crypto Provider";
+            const string container = "{123ABCDE-1234-ABCD-1234-ABCD1234ABCD}";
+
+            Console.Write($"Enter {(correctPassword ? "correct" : "incorrect")} password: ");
+
+            using (var pwd = ReadPassword())
+            {
+                var cspParameters = new CspParameters(1, provider, container)
+                {
+                    KeyNumber = (int)KeyNumber.Exchange,
+                    Flags = CspProviderFlags.UseExistingKey | CspProviderFlags.UseMachineKeyStore,
+                    KeyPassword = pwd
+                };
+
+                Action sign = () =>
+                {
+                    using (var rsa = new RSACryptoServiceProvider(cspParameters))
+                    {
+                        var signed = rsa.SignData(new byte[3] { 1, 2, 3 }, "sha256");
+                        Console.WriteLine(Convert.ToBase64String(signed));
+                    }
+                };
+
+                if (correctPassword)
+                    sign();
+                else
+                    Assert.ThrowsAny<CryptographicException>(sign);
+            }
+        }
+
+        static SecureString ReadPassword()
+        {
+            var ret = new SecureString();
+
+            while (true)
+            {
+                ConsoleKeyInfo c = Console.ReadKey(true);
+                if (c.Key == ConsoleKey.Enter)
+                {
+                    Console.WriteLine();
+                    break;
+                }
+                else if (c.Key == ConsoleKey.Escape)
+                {
+                    for (int i = 0; i < ret.Length; i++)
+                        Console.Write("\b \b");
+
+                    ret.Clear();
+                }
+                else if (c.Key == ConsoleKey.Backspace)
+                {
+                    if (ret.Length > 0)
+                    {
+                        Console.Write("\b \b");
+                        ret.RemoveAt(ret.Length - 1);
+                    }
+                }
+                else if (c.Key == ConsoleKey.LeftArrow
+                    || c.Key == ConsoleKey.RightArrow
+                    || c.Key == ConsoleKey.DownArrow
+                    || c.Key == ConsoleKey.UpArrow)
+                {
```

Comment:
My college roommate would be saddened by this.  He had passwords he could only type with the left and right arrow keys to change the character position. :smile:

This doesn't seem needed, particularly in uncalled/uncallable test code.  Can we trim down this method? (And if we're keeping it, add the access modifier?)